### PR TITLE
[IMP] mail: disable composer textarea when posting

### PR DESCRIPTION
**Pull request by phenix-factory**
task-2390637

### CODE REVIEW

**Comment by phenix-factory** [comment]
I have try to create a test for this. But I guess that the change is so fast that I can't catch when the `textarea` is `readonly`.

**Comment by seb-odoo** [comment]
You can mock message post rpc to control it's resolve time.

File: addons/mail/static/src/components/composer/composer_tests.js [hunk 1]
```diff
@@ -2098,6 +2099,45 @@ QUnit.test("mentioned partners should not be notified if they are not member of
...
+        },
+    });
+    const thread = this.env.models['mail.thread'].find(thread =>
+        thread.id === 20 &&
+        thread.model === 'mail.channel'
+    );
+    await this.createComposerComponent(thread.composer);
+    // Type message
+    await afterNextRender(() => {
+        document.querySelector(`.o_ComposerTextInput_textarea`).focus();
```
Comment (alexkuhn) [inline addons/mail/static/src/components/composer/composer_tests.js:2125]: use single quotes that backquotes

File: addons/mail/static/src/components/composer/composer_tests.js [hunk 2]
```diff
@@ -2098,6 +2099,45 @@ QUnit.test("mentioned partners should not be notified if they are not member of
...
+    });
+
+    // Send message
+    await afterNextRender(() =>
+        document.querySelector('.o_Composer_buttonSend').click()
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ComposerTextInput_textarea[readonly]',
+        "Composer readonly test"
```
Comment (alexkuhn) [inline addons/mail/static/src/components/composer/composer_tests.js:2136]: Bad assertion comment. State precisely that the text input should be readonly when message is being posted.

File: addons/mail/static/src/components/composer/composer_tests.js [hunk 3]
```diff
@@ -2098,6 +2099,45 @@ QUnit.test("mentioned partners should not be notified if they are not member of
...
+    );
+    messagePostDef.resolve();
```
Comment (alexkuhn) [inline addons/mail/static/src/components/composer/composer_tests.js:2138]: Useless statement.

**Review by alexkuhn — CHANGES_REQUESTED** [review]
Not just the textarea should be prevented changes during message post: there's also add attachment, interactions on attachments, composer header suggested recipients, emoji button, etc. Maybe even possibly force not showing the mention suggestion list (not sure, but could be displayed before posting, should ensure click on button doesn't keep it displayed).

**Comment by alexkuhn** [comment]
Also change `[IMP]` to `[FIX]`, and resolve conflicting files during rebase.

**Pull request closed without merging**